src/ROOTNTupleReader.cc

```diff
@@ -0,0 +1,245 @@
+#include "podio/ROOTNTupleReader.h"
+#include "podio/CollectionBase.h"
+#include "podio/CollectionBuffers.h"
+#include "podio/CollectionIDTable.h"
+#include "podio/GenericParameters.h"
+#include "rootUtils.h"
+
+#include "TClass.h"
+#include <ROOT/RError.hxx>
+#include <memory>
+
+namespace podio {
+
+GenericParameters ROOTNTupleReader::readEventMetaData(const std::string& name, unsigned entNum) {
```

Comment:
I think this could be made less repetitive, via a templated function that handles one type, the help of `std::apply` and the `SupportedGenericDataTypes`. That would require to make the `getMap` methods public in `GenericParameters`, but they effectively are already.

Reply:
Implemented